### PR TITLE
Fixes #22514 - Added example for using image digest in the docker run command

### LINKED ISSUE #22514: `docker run` documentation could be improved with respect to using digests
I had some trouble working out the correct way to use `docker run` with the sha256 digest. I found the documentation a little confusing with regard to this. For example the [run reference page][ref] has the following example:

[ref]: https://docs.docker.com/engine/reference/run

```
$ docker run [OPTIONS] IMAGE[:TAG|@DIGEST] [COMMAND] [ARG...]
```
However I had problems trying to run my image this way. For example:

``` bash
docker run namespace/image@digest # Wrong - parse error
docker run namespace/image@sha256:digest # Correct
```
I had to work this out from trial and error. The link to the [Image[@digest]][digest] section anchor doesn't have an example, which I think might have been useful. I believe therefore it might be beneficial to update the documentation regarding using digests.

I suggest the **General Form** section to be:

[digest]: https://docs.docker.com/engine/reference/run/#image-digest

```
$ docker run [OPTIONS] IMAGE[:TAG|@sha256:DIGEST] [COMMAND] [ARG...]
```
If there are other possible digests, then sha256 could be changed to DIGEST_TYPE or something similar.

Secondly I suggest that the [Image[@digest]][digest] anchor have the following example of using the digest:

```
docker run alpine@sha256:9cacb71397b640eca97488cf08582ae4e4068513101088e9f96c9814bfda95e0 date
```

This is my feedback from initially trying to get digests to run with `docker run`. If this was confusing to me, it might possibly be confusing to other users.

### PR DESCRIPTION
**Pull request by sbose78**
**\- What I did**
Added a sample `docker run` command which uses the image digest

**\- How I did it**
updated `docs/reference/run.md` with the relevant markup

**\- How to verify it**
Run a `make docs` and navigate to `http://<hostname-of-dev->:8000/engine/reference/run/`
and verify the "Image[@digest]" section

**\- Description for the changelog**
Fixes #22514 - Added a sample `docker run` command which uses the image digest

### CODE REVIEW

File: docs/reference/run.md [hunk 1]
```diff
@@ -190,6 +190,8 @@ Images using the v2 or later image format have a content-addressable identifier
 called a digest. As long as the input used to generate the image is unchanged,
 the digest value is predictable and referenceable.
 
+    docker run alpine@sha256:9cacb71397b640eca97488cf08582ae4e4068513101088e9f96c9814bfda95e0 date
```
Comment (thaJeztah) [inline docs/reference/run.md]: Thanks! Can you add a short explanation of the example itself? Something like;

```markdown
The following example runs a container from the `alpine` image
with the `sha256:9cacb71397b640eca97488cf08582ae4e4068513101088e9f96c9814bfda95e0`
digest:

    $ docker run alpine@sha256:9cacb71397b640eca97488cf08582ae4e4068513101088e9f96c9814bfda95e0 date
```

Note that for shell examples, we alway include the prompt (`$ `)

**Comment by sbose78** [comment]
Thanks @thaJeztah  , made the change. Please review.

File: docs/reference/run.md [hunk 2]
```diff
@@ -190,6 +190,10 @@ Images using the v2 or later image format have a content-addressable identifier
 called a digest. As long as the input used to generate the image is unchanged,
 the digest value is predictable and referenceable.
 
+The following example runs a container from the `alpine` image with the `sha256:9cacb71397b640eca97488cf08582ae4e4068513101088e9f96c9814bfda95e0` digest:
```
Comment (thaJeztah) [inline docs/reference/run.md]: Can you wrap this to ~80 chars?

**Comment by thaJeztah** [comment]
Thanks! Sorry, one more nit, and can you also squash your commits, so that there's a single commit in your PR? You can find some information on that in our contributors guide; https://docs.docker.com/opensource/workflow/work-issue/#pull-and-rebase-frequently

**Comment by sbose78** [comment]
Thanks!
I've squashed commits.
But there's no way I would be able to compress the content within 80 words :( the digest itself is 60+
Or, am I missing something here ?

**Comment by thaJeztah** [comment]
Oh, it doesn't have to be compressed to 80chars, just a newline to split it over multiple lines (not in the middle of the digest of course).

Anyway, it's just a nit, so no need to update.

LGTM

**Comment by sbose78** [comment]
Oh. 
Thanks @thaJeztah !

**Comment by thaJeztah** [comment]
To clarify; wrapping to 80 chars makes it easier to review, and easier to read when viewing the source code, so it's not a big issue, but we try to stick to that where possible

**Comment by vdemeester** [comment]
LGTM 🐮

**Comment by sbose78** [comment]
Thanks @vdemeester  and @thaJeztah , I've incorporated the wrapping.

**Comment by thaJeztah** [comment]
awesome! thanks so much!

**Pull request merged**